src/main/java/space/obminyashka/items_exchange/service/impl/RefreshTokenServiceImpl.java

```diff
@@ -21,14 +20,16 @@ public class RefreshTokenServiceImpl implements RefreshTokenService {
     private final RefreshTokenRepository refreshTokenRepository;
 
     @Override
-    public RefreshToken createRefreshToken(User user) {
-        final String token = jwtTokenService.generateRefreshToken(user.getUsername());
+    public RefreshToken createRefreshToken(String createdRefreshToken, String username) {
+        final String token = jwtTokenService.generateRefreshToken(username);
         final var tokenExpirationTime = jwtTokenService.generateRefreshTokenExpirationTime();
-        return refreshTokenRepository.save(Optional.ofNullable(user.getRefreshToken())
-                .map(refreshToken -> refreshToken.setToken(token)
-                        .setCreated(LocalDateTime.now())
-                        .setExpiryDate(tokenExpirationTime))
-                .orElseGet(() -> new RefreshToken(user, token, tokenExpirationTime)));
+
+        Optional.ofNullable(createdRefreshToken)
+                .ifPresentOrElse(
+                        (refreshToken) -> refreshTokenRepository.updateRefreshToken(username, token, tokenExpirationTime),
+                        () -> refreshTokenRepository.createRefreshToken(username, token, tokenExpirationTime));
```

Comment:
You don't use any Optional feature instead of ifPresentOrElse, it would be mush simpler to rewrite it with if/else expression